htdocs/js/ui/notebook_protection.js

```diff
@@ -1,88 +1,113 @@
 RCloud.UI.notebook_protection = (function() {
 
-    this.passedData = null;
-    this.appInited = false;
-    this.theApp = null;
-
-
     //set from outside
+    this.defaultCryptogroup = null;
+    this.defaultNotebook = null;
     this.userId;
     this.userLogin;
```

Comment:
for sure, all of those types of declarations should be initted with either null, '', [] or {}.